src/ORM/ArrayList.php

```diff
@@ -638,28 +622,102 @@ public function filter()
      *
+     * Also supports SearchFilter syntax
+     * @example // include anyone with "sam" anywhere in their name
+     *          $list = $list->filterAny('Name:PartialMatch', 'sam');
+     *
      * @param string|array See {@link filter()}
      * @return static
      */
     public function filterAny()
     {
-        $keepUs = $this->normaliseFilterArgs(...func_get_args());
+        $filters = call_user_func_array([$this, 'normaliseFilterArgs'], func_get_args());
+        return $this->filterOrExclude($filters, true, true);
+    }
 
+    /**
+     * Exclude the list to not contain items with these characteristics
+     *
+     * @return ArrayList
+     * @see SS_List::exclude()
+     * @example $list->exclude('Name', 'bob'); // exclude bob from list
+     * @example $list->exclude('Name', array('aziz', 'bob'); // exclude aziz and bob from list
+     * @example $list->exclude(array('Name'=>'bob, 'Age'=>21)); // exclude bob that has Age 21
+     * @example $list->exclude(array('Name'=>'bob, 'Age'=>array(21, 43))); // exclude bob with Age 21 or 43
+     * @example $list->exclude(array('Name'=>array('bob','phil'), 'Age'=>array(21, 43)));
+     *          // bob age 21 or 43, phil age 21 or 43 would be excluded
+     *
+     * Also supports SearchFilter syntax
+     * @example // everyone except anyone with "sam" anywhere in their name
+     *          $list = $list->exclude('Name:PartialMatch', 'sam');
+     */
+    public function exclude()
+    {
+        $filters = call_user_func_array([$this, 'normaliseFilterArgs'], func_get_args());
+        return $this->filterOrExclude($filters, false);
+    }
+
+    /**
+     * Return a copy of the list excluding any items that have any of these characteristics
+     *
+     * @example // everyone except bob in the list
+     *          $list = $list->excludeAny('Name', 'bob');
+     * @example // everyone except azis or bob in the list
+     *          $list = $list->excludeAny('Name', array('aziz', 'bob');
+     * @example // everyone except bob or anyone aged 21 in the list
+     *          $list = $list->excludeAny(array('Name'=>'bob, 'Age'=>21));
+     * @example // everyone except bob or anyone aged 21 or 43 in the list
+     *          $list = $list->excludeAny(array('Name'=>'bob, 'Age'=>array(21, 43)));
+     * @example // everyone except all bobs, phils or anyone aged 21 or 43 in the list
+     *          $list = $list->excludeAny(array('Name'=>array('bob','phil'), 'Age'=>array(21, 43)));
+     *
+     * Also supports SearchFilter syntax
+     * @example // everyone except anyone with "sam" anywhere in their name
+     *          $list = $list->excludeAny('Name:PartialMatch', 'sam');
+     *
+     * @param string|array See {@link filter()}
+     */
+    public function excludeAny(): static
+    {
+        $filters = call_user_func_array([$this, 'normaliseFilterArgs'], func_get_args());
+        return $this->filterOrExclude($filters, false, true);
+    }
```

Comment:
Adding this isn't _necessary_ by any means, but it's a very low effort low risk change that helps align the ArrayList API with other lists. I think it's well worth adding.